nucypher/network/nodes.py

```diff
@@ -943,6 +943,7 @@ class Teacher:
     TEACHER_VERSION = LEARNING_LOOP_VERSION
     _interface_info_splitter = (int, 4, {'byteorder': 'big'})
     log = Logger("teacher")
+    synchronous_query_timeout = 20  # How long to wait during REST endpoints for blockchain queries to resolve
```

Comment:
Well, we need to have a reasonable timeout time (and eventually, probably a rethinking of this round-trip from Alice to Ursula).  Why do we imagine that it will take longer than 20 seconds to read a record from the blockchain?  Is that really a thing?  (Asking honestly, I don't know).

To me, the flaw here is that Alice probably needs to wait for confirmations to ensure that the Policy is even in a block.  But we can fix that down the road.

@KPrasch's answer to the right number of seconds for this, off the cuff, was 10.  But I mean, yeah, it needs consideration.